### PR TITLE
Enable support for unique names in text search with id: prefix

### LINKED ISSUE #63: Searching for id:unique_name not supported

### PR DESCRIPTION
**Pull request by robvandenbogaard**
Fix #63
Fix https://github.com/driebit/ginger/issues/591

### CODE REVIEW

File: support/elasticsearch_search.erl [hunk 1]
```diff
@@ -482,10 +482,15 @@ map_must({rsc_id, Id}, _Context) ->
...
     map_must({text, list_to_binary(Val)}, Context);
-map_must({text, <<"id:", Id/binary>>}, _Context) ->
-    %% Find by id when text argument equals "id:123"
+map_must({text, <<"id:", IdOrName/binary>>}, Context) ->
+    %% Find by id when text argument equals "id:123" or "id:unique_name"
+    TrimmedId = z_string:trim(IdOrName),
+    Id = case m_rsc:name_to_id(TrimmedId, Context) of
+              {ok, NumericId} -> integer_to_binary(NumericId);
+              _ -> TrimmedId
+          end,
```
Comment (ddeboer) [inline support/elasticsearch_search.erl]: You can use `Id = m_rsc:rid(IdOrName, Context)` instead.

Comment (robvandenbogaard) [reply support/elasticsearch_search.erl]: Ah yes much better 👍 will do

File: support/elasticsearch_search.erl [hunk 2]
```diff
@@ -482,10 +482,11 @@ map_must({rsc_id, Id}, _Context) ->
...
 map_must({text, "id:" ++ _ = Val}, Context) ->
     map_must({text, list_to_binary(Val)}, Context);
-map_must({text, <<"id:", Id/binary>>}, _Context) ->
-    %% Find by id when text argument equals "id:123"
+map_must({text, <<"id:", IdOrName/binary>>}, Context) ->
+    %% Find by id when text argument equals "id:123" or "id:unique_name"
+    Id = m_rsc:rid(z_string:trim(IdOrName), Context),
     {true, #{<<"match">> => #{
-        <<"_id">> => z_string:trim(Id)
+        <<"_id">> => Id
```
Comment (ddeboer) [inline support/elasticsearch_search.erl]: And then, of course, we can just ditch the variable `Id`. 😉

Comment (robvandenbogaard) [reply support/elasticsearch_search.erl]: Hmm yes - somehow I liked it to have its own line, but indeed.

**Pull request merged**